### ISSUE TITLE
dnsdist: segfault in beta1

### ISSUE THREAD

**Issue by rubenk** [PowerDNS/pdns#3716]
Using the following config:
```
newServer({"127.0.0.1", pool="local"})
 
pc = newPacketCache(10000)
getPool("local"):setCache(pc)
```

segfaults.

```
Read configuration from '/etc/dnsdist/dnsdist.conf'

Program received signal SIGSEGV, Segmentation fault.
std::rethrow_exception (ep=...) at ../../../../libstdc++-v3/libsupc++/eh_ptr.cc:241
241	  __atomic_add_fetch (&eh->referenceCount, 1,  __ATOMIC_ACQ_REL);
(gdb) thread apply all bt

Thread 1 (Thread 0x7ffff7f5a8c0 (LWP 10767)):
#0  std::rethrow_exception (ep=...) at ../../../../libstdc++-v3/libsupc++/eh_ptr.cc:241
#1  0x000055555558b9bc in LuaContext::callRaw (state=0x40000378, functionAndArguments=..., outArguments=0) at ext/luawrapper/include/LuaContext.hpp:1348
#2  0x0000555555605a03 in LuaContext::call<std::tuple<>>(lua_State*, LuaContext::PushedObject) (state=0x40000378, toCall=...) at ext/luawrapper/include/LuaContext.hpp:1317
#3  0x00005555555cdb25 in LuaContext::executeCode (code=..., this=<optimized out>) at ext/luawrapper/include/LuaContext.hpp:221
#4  setupLua (client=<optimized out>, config="/etc/dnsdist/dnsdist.conf") at dnsdist-lua.cc:1397
#5  0x000055555556d59f in main (argc=<optimized out>, argv=<optimized out>) at dnsdist.cc:1522
```

**Comment by rgacogne** [comment]
Did you get this using our packages? On which distribution? I don't manage to get anything else than 'Fatal error: invalid key to 'next' with this configuration for now.

**Comment by pieterlexis** [comment]
closed due to lack of response

**Comment by rubenk** [comment]
If I recall correctly, this was the Fedora rpm.

I can still reproduce it with dnsdist.x86_64 1.0.0-1.fc23

```
(gdb) r
Starting program: /usr/bin/dnsdist 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Read configuration from '/etc/dnsdist/dnsdist.conf'

Program received signal SIGSEGV, Segmentation fault.
std::rethrow_exception (ep=...) at ../../../../libstdc++-v3/libsupc++/eh_ptr.cc:241
241	  __atomic_add_fetch (&eh->referenceCount, 1,  __ATOMIC_ACQ_REL);
(gdb) thread apply all backtrace

Thread 1 (Thread 0x7ffff7f538c0 (LWP 1559)):
#0  std::rethrow_exception (ep=...) at ../../../../libstdc++-v3/libsupc++/eh_ptr.cc:241
#1  0x000055555558ba9c in LuaContext::callRaw (state=0x40000378, functionAndArguments=..., outArguments=0) at ext/luawrapper/include/LuaContext.hpp:1348
#2  0x0000555555606033 in LuaContext::call<std::tuple<>>(lua_State*, LuaContext::PushedObject) (state=0x40000378, toCall=...) at ext/luawrapper/include/LuaContext.hpp:1317
#3  0x00005555555cdca8 in LuaContext::executeCode (code=..., this=<optimized out>) at ext/luawrapper/include/LuaContext.hpp:221
#4  setupLua (client=<optimized out>, config="/etc/dnsdist/dnsdist.conf") at dnsdist-lua.cc:1401
#5  0x000055555556d66f in main (argc=<optimized out>, argv=<optimized out>) at dnsdist.cc:1522
```

**Comment by rgacogne** [comment]
Thank you! Issue reproduced on FC 23 with LuaJIT. It does not happen with Lua though.

**Comment by rgacogne** [comment]
Fixed by #4004.
